docs/standard/native-interop/preserve-sig.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Implicit native signature translations and PreserveSig in .NET native interop
+description: Learn how managed signatures translate to native signatures for interop methods in .NET
+ms.date: 9/21/2023
+---
+
+# Implicit method signature translations in .NET interop
+
+In order to stay programming language agnostic, the Windows COM system and many Windows APIs return a 4 byte integer type called an `HRESULT` to indicate whether an API succeeded or failed, along with some information about the failure. Other values that need to be passed to the caller are "returned" via pointer parameters that act as "out" parameters, and are typically the last parameter in the signature. Languages like C# and Visual Basic traditionally translate a failure code to an exception to match how failures are usually propagated in the language, and the "out" parameter was treated as the return value in the signature of the managed language representation. For example, see below a native COM method signature and its corresponding C# signature.
```

Comment:
```suggestion
In order to stay programming language agnostic, the Windows COM system and many Windows APIs return a 4 byte integer type called an `HRESULT` to indicate whether an API succeeded or failed, along with some information about the failure. Other values that need to be passed to the caller are "returned" via pointer parameters that act as "out" parameters, typically as the last parameter in the signature. Languages like C# and Visual Basic traditionally translate a failure code to an exception to match how failures are usually propagated in the language&mdash;the "out" parameter is treated as the return value in the signature of the managed language representation. For example, see the following native COM method signature and its corresponding C# signature:
```

Reply:
I completely reworded this paragraph to frame it more as a translation from C# to native rather than native to C#.